src/models/blockchain/transaction/utxo.rs

```diff
@@ -163,7 +163,7 @@ impl Utxo {
 
     /// If the UTXO has a timelock, find out what the release date is.
     pub fn release_date(&self) -> Option<Timestamp> {
-        self.coins.iter().filter_map(Coin::release_date).next()
+        self.coins.iter().find_map(Coin::release_date)
```

Comment:
Might be a bug here (unrelated to your change), we probably want to take the "biggest" release data here, in case URXO has multiple instances of the same timelock type script.